src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/document/DocumentManagementServiceTest.java

```diff
@@ -0,0 +1,183 @@
+                "http://localhost:8080/documents/0fa1ab60-f836-43aa-8c65-b07cc9bebcbe"
+            );
+
+        assertThat(actualUploadResponse)
+            .extracting("fileName")
+            .containsExactly(
+                "test1.pdf", "test2.pdf"
+            );
+
+        verify(authTokenGenerator).generate();
+        verify(restTemplate).postForObject(
+            eq("http://localhost:8080/documents"),
+            httpEntityReqEntity.capture(),
+            any()
+        );
+    }
+
+    @Test
+    public void should_throw_client_exception_when_service_auth_throws_unauthorized_exception() throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("test2.pdf"));
+
+        Pdf pdf1 = new Pdf("test1.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("test2.pdf", test2PdfBytes);
+
+        when(authTokenGenerator.generate()).thenThrow(new HttpClientErrorException(HttpStatus.UNAUTHORIZED));
+
+        //when
+        Throwable exc = catchThrowable(() -> documentManagementService.uploadDocuments(asList(pdf1, pdf2)));
+
+        //then
+        assertThat(exc)
+            .isInstanceOf(HttpClientErrorException.class);
+
+        verify(authTokenGenerator).generate();
+    }
+
+    @Test
+    public void should_throw_unable_to_upload_doc_exception_when_bulk_scan_service_throws_client_exception()
+        throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("test2.pdf"));
+
+        Pdf pdf1 = new Pdf("test1.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("test2.pdf", test2PdfBytes);
+
+        when(restTemplate.postForObject(
+            eq("http://localhost:8080/documents"),
+            httpEntityReqEntity.capture(), any())
+        ).thenThrow(new HttpClientErrorException(HttpStatus.FORBIDDEN));
+
+        //when
+        Throwable exc = catchThrowable(() -> documentManagementService.uploadDocuments(asList(pdf1, pdf2)));
+
+        //then
+        assertThat(exc)
+            .isInstanceOf(UnableToUploadDocumentException.class)
+            .hasCauseExactlyInstanceOf(HttpClientErrorException.class);
+
+        verify(authTokenGenerator).generate();
+    }
+
+    @Test
+    public void should_throw_unable_to_upload_document_exception_when_document_storage_is_down() throws Exception {
+        //Given
+        byte[] test1PdfBytes = toByteArray(getResource("test1.pdf"));
+        byte[] test2PdfBytes = toByteArray(getResource("test2.pdf"));
+
+        Pdf pdf1 = new Pdf("test1.pdf", test1PdfBytes);
+        Pdf pdf2 = new Pdf("test2.pdf", test2PdfBytes);
+
+        when(restTemplate.postForObject(
+            eq("http://localhost:8080/documents"),
+            httpEntityReqEntity.capture(),
+            any())
+        ).thenThrow(new HttpServerErrorException(HttpStatus.INTERNAL_SERVER_ERROR));
```

Comment:
you can use methods from `BDDMockito` class, that will better match given-when-then convention that you're using in these tests.